Sources/SpeziViews/Views/Layout/HorizontalGeometryReader.swift

```diff
@@ -40,9 +41,19 @@ public struct HorizontalGeometryReader<Content: View>: View {
                 }
             )
             .onPreferenceChange(WidthPreferenceKey.self) { width in
-                runOrScheduleOnMainActor {
-                    self.width = width
+                if Thread.isMainThread {
+                    MainActor.assumeIsolated {
+                        self.width = width
+                    }
+                } else {
+                    values.continuation.yield(width)
+                }
+            }
+            .task {
+                for await value in values.stream {
+                    self.width = value
                 }
+                values = AsyncStream.makeStream()
```

Comment:
@bauer-andreas why do we re-initialize the stream at this point? given the fact that we never call `continuation.finish()`, isn't this practically unreachable? or am i missing something here?